tests/Aspire.Hosting.Testing.Tests/TestingFactoryTests.cs

```diff
@@ -47,6 +48,9 @@ public async Task HttpClientGetTest()
         var rns = _app.Services.GetRequiredService<ResourceNotificationService>();
         await rns.WaitForResourceAsync("mywebapp1").WaitAsync(TimeSpan.FromSeconds(60));
 
+        // Wait for the application to be ready
+        await _app.WaitForTextAsync("Application started.").WaitAsync(TimeSpan.FromMinutes(1));
```

Comment:
We are already `await rns.WaitForResourceAsync("mywebapp1")` above. Why is this wait needed?

If the existing code isn't good enough, then i think we might need to change our test template code as well, because the previous code is basically what the template spits out:

https://github.com/dotnet/aspire/blob/024b48bf581f6790b0878128c6acff7863d3d5f7/src/Aspire.ProjectTemplates/templates/aspire-xunit/IntegrationTest1.cs#L25-L32

cc @DamianEdwards @ReubenBond

Reply:
I assume `WaitForResourceAsync` doesn't account for when the service can actually serve requests. I dumped the logs for this test and added markers before/after each call:

```
2024-08-08T18:39:12.3897554Z WaitForResourceAsync - Start
2024-08-08T18:39:13.5023642Z WaitForResourceAsync - Stopped
2024-08-08T18:39:13.5024903Z WaitForTextAsync - Start
2024-08-08T18:39:14.3305581Z WaitForTextAsync - Stopped
```

So quite some delay between WaitForResourceAsync and the application is ready.

Reply:
My concern is how does the tests in our template work today without this? Do we hve this flakiness issue in them?